pkg/xray/watches.go

```diff
@@ -196,34 +218,21 @@ func resourceXrayWatchCreate(ctx context.Context, d *schema.ResourceData, m inte
 }
 
 func resourceXrayWatchRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	watch := Watch{}
-	resp, err := m.(*resty.Client).R().SetResult(&watch).Get("xray/api/v2/watches/" + d.Id())
+	watch, resp, err := getWatch(d.Id(), m.(*resty.Client))
 	if err != nil {
 		if resp != nil && resp.StatusCode() == http.StatusNotFound {
 			log.Printf("[WARN] Xray watch (%s) not found, removing from state", d.Id())
 			d.SetId("")
 		}
 		return diag.FromErr(err)
 	}
-	if err := d.Set("description", watch.GeneralData.Description); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := d.Set("active", watch.GeneralData.Active); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := d.Set("watch_resource", packProjectResources(watch.ProjectResources)); err != nil {
-		return diag.FromErr(err)
-	}
-	if err := d.Set("assigned_policy", packAssignedPolicies(watch.AssignedPolicies)); err != nil {
-		return diag.FromErr(err)
-	}
-
+	packWatch(watch, d)
 	return nil
 }
 
 func resourceXrayWatchUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	watch := unpackWatch(d)
-	resp, err := m.(*resty.Client).R().SetBody(watch).Put("xray/api/v2/watches/" + d.Id())
+	resp, err := m.(*resty.Client).R().AddRetryCondition(retryOnMergeError).SetBody(watch).Put("xray/api/v2/watches/" + d.Id())
```

Comment:
Why all the retry code?

Reply:
@alexhung asked me to add the retry to avoid the problems RT had.

Reply:
@alexhung - I am of the mind that we shouldn't hand an error unless we know what it means. This seems to be an RT only error. So, in this case, it'd be useless to retry here. But, if it did occur, I'd like to know what that meant. Because, when I've seen this error, it doesn't resolve until the repo is created. Meaning: xray should never see it

Reply:
@chb0github Fair comment. We know Artifactory/Xray API has rate limits so we should handle that regardless of if our users find it themselves. If all we need to handle is 429 then let's do that and call it done.

@danielmkn did you setup your environment to trigger the need to retry, and note the type of status code/error message we need to handle?

Reply:
no

Reply:
@danielmkn Please do that so we know what we need to handle. My mistake in approving the PR assuming you've done this verification.

Reply:
@alexhung - then, let's handle 429. This error is not code specific at all. In fact, I hate it because it's based on the body payload which they could change. That's why I say: Handle the error when we see it - preferrably we see it first.